### PR TITLE
feat: Support arbitrary module namespace names in no-restricted-imports

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[x] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

Refs #15465, fixes `no-restricted-imports`.

#### What changes did you make? (Give an overview)

Updated the `no-restricted-imports` rule to correctly read quoted module export names.

Before this change:

```js
/*eslint no-restricted-imports: ["error", { paths: [{
    name: "mod",
    importNames: ["foo"]    
}]}]*/

import { "foo" as bar } from "mod"; // ok, false negative

export { "foo" } from "mod"; // ok, false negative

export { "foo" as bar } from "mod"; // ok, false negative
```

After this change:

```js
/*eslint no-restricted-imports: ["error", { paths: [{
    name: "mod",
    importNames: ["foo"]    
}]}]*/

import { "foo" as bar } from "mod"; // error

export { "foo" } from "mod"; // error

export { "foo" as bar } from "mod"; // error
```

### CODE REVIEW

**Review by aladdin-add — APPROVED** [review]

**Review by nzakas — APPROVED** [review]
LGTM. Would like another set of eyes before merging.

**Comment by mdjermanovic** [comment]
Merging as there are two approvals.

**Pull request merged**
